pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -53,35 +53,49 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 		return nil
 	}
 
-	// Check if the latest deployment already exists
-	if deployment, err := c.deploymentClient.getDeployment(config.Namespace, deployutil.LatestDeploymentNameForConfig(config)); err != nil {
-		if !errors.IsNotFound(err) {
-			return fmt.Errorf("couldn't get deployment for config %s: %v", deployutil.LabelForDeploymentConfig(config), err)
-		}
-	} else {
-		// If there's an existing deployment, nothing needs to be done.
-		if deployment != nil {
-			return nil
-		}
-	}
-
-	// Check if any previous deployment is still running (any non-terminal state).
+	// Check if any existing inflight deployments (any non-terminal state).
 	existingDeployments, err := c.deploymentClient.listDeploymentsForConfig(config.Namespace, config.Name)
 	if err != nil {
 		return fmt.Errorf("couldn't list deployments for config %s: %v", deployutil.LabelForDeploymentConfig(config), err)
 	}
+	var inflightDeployment, deploymentForCancellation *kapi.ReplicationController
 	for _, deployment := range existingDeployments.Items {
 		deploymentStatus := deployutil.DeploymentStatusFor(&deployment)
 		switch deploymentStatus {
 		case deployapi.DeploymentStatusFailed,
 			deployapi.DeploymentStatusComplete:
 			// Previous deployment in terminal state - can ignore
-			// Ignoring specific deployment states so that any new
+			// Ignoring specific deployment states so that any newly introduced
 			// deployment state will not be ignored
 		default:
-			glog.V(4).Infof("Found previous deployment %s (status %s) - will requeue", deployutil.LabelForDeployment(&deployment), deploymentStatus)
-			return transientError(fmt.Sprintf("found previous deployment (state: %s) for %s - requeuing", deploymentStatus, deployutil.LabelForDeploymentConfig(config)))
+			if inflightDeployment == nil {
+				inflightDeployment = &deployment
```

Comment:
The `default` block of the switch statement goes through (at least this is what the code is supposed to do) all inflight (any non-terminal state) deployments and cancels all older deployments, leaving just the latest one (based on latest config version).